benchmark/Cajita_FastFourierTransformPerformance.cpp

```diff
@@ -0,0 +1,240 @@
+                   comm );
+    outputResults( stream, "grid_size_per_dim", grid_sizes_per_dim,
+                   transforms_timer, comm );
+
+    stream << std::flush;
+}
+
+//---------------------------------------------------------------------------//
+// main
+int main( int argc, char* argv[] )
+{
+    // Initialize environment
+    MPI_Init( &argc, &argv );
+    Kokkos::initialize( argc, argv );
+
+    // Check arguments.
+    if ( argc < 2 )
+        throw std::runtime_error( "Incorrect number of arguments. \n \
+             First argument - file name for output \n \
+             \n \
+             Example: \n \
+             $/: ./FastFourierTransformPerformance test_results.txt\n" );
+
+    // Define run sizes.
+    std::string run_type = "";
+    if ( argc > 2 )
+    {
+        run_type = argv[2];
+    }
+
+    // Declare the grid size per dimension
+    // currently, testing 3dims+symmetric
+    std::vector<double> grid_sizes_per_dim = { 10, 20 };
+    if ( run_type == "large" )
+    {
+        grid_sizes_per_dim = { 100, 200 };
+    }
+
+    // Get the name of the output file.
+    std::string filename = argv[1];
+
+    // Barrier before continuing
+    MPI_Barrier( MPI_COMM_WORLD );
+
+    // Get comm rank;
+    int comm_rank;
+    MPI_Comm_rank( MPI_COMM_WORLD, &comm_rank );
+
+    // Open the output file on rank 0.
+    std::fstream file;
+    if ( 0 == comm_rank )
+        file.open( filename, std::fstream::out );
+
+    // Output file header
+    if ( 0 == comm_rank )
+    {
+        file << "\n";
+        file << "Cajita FFT Performance Benchmark"
+             << "\n";
+        file << "----------------------------------------------"
+             << "\n";
+        file << std::flush;
+    }
+
+    // Run the tests.
+#ifdef KOKKOS_ENABLE_SERIAL
+    using SerialDevice = Kokkos::Device<Kokkos::Serial, Kokkos::HostSpace>;
+#ifdef Heffte_ENABLE_FFTW
+    performanceTest<SerialDevice, Experimental::FFTBackendFFTW>(
+        file, grid_sizes_per_dim, MPI_COMM_WORLD, "serial_fftw_" );
+#endif
+#ifdef Heffte_ENABLE_MKL
+    performanceTest<SerialDevice, Experimental::FFTBackendMKL>(
+        file, grid_sizes_per_dim, MPI_COMM_WORLD, "serial_mkl_" );
+#endif
+#endif
+#ifdef KOKKOS_ENABLE_OPENMP
+    using OpenMPDevice = Kokkos::Device<Kokkos::OpenMP, Kokkos::HostSpace>;
+#ifdef Heffte_ENABLE_FFTW
+    performanceTest<OpenMPDevice, Experimental::FFTBackendFFTW>(
+        file, grid_sizes_per_dim, MPI_COMM_WORLD, "openmp_fftw_" );
+#endif
+#ifdef Heffte_ENABLE_MKL
+    performanceTest<OpenMPDevice, Experimental::FFTBackendMKL>(
+        file, grid_sizes_per_dim, MPI_COMM_WORLD, "openmp_mkl_" );
+#endif
+#endif
+
+#ifdef KOKKOS_ENABLE_CUDA
+    using CudaDevice = Kokkos::Device<Kokkos::Cuda, Kokkos::CudaSpace>;
+#ifdef Heffte_ENABLE_CUDA
+    performanceTest<CudaDevice, Experimental::Impl::FFTBackendDefault>(
+        file, grid_sizes_per_dim, MPI_COMM_WORLD, "openmp_cuda_" );
```

Comment:
Looks like a copy/paste label mistake here and for HIP

Reply:
I'm not sure? I did switch the order of ifdefs here, just to match the ifdef ordering for other backends. It's not a necessary change, though. Very possible I'm just not seeing what is wrong though.

Reply:
Ah, thanks! I need my vision checked